flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/RecordWriterOutput.java

```diff
@@ -140,7 +146,8 @@ public void emitLatencyMarker(LatencyMarker latencyMarker) {
     }
 
     public void broadcastEvent(AbstractEvent event, boolean isPriorityEvent) throws IOException {
-        recordWriter.broadcastEvent(event, isPriorityEvent);
+        recordWriter.broadcastEvent(
+                event, isPriorityEvent && !(event instanceof CheckpointBarrier && isPointwise));
```

Comment:
I think this is not enough. This way you are sending potentially an unaligned or timeoutable aligned CB as non priority event, but this priority flag only has an effect on the output. On the input this could still be processed as UC or timeout to UC?

Unless you think it's fine, and we can rescale from pointwise connections if UC is supported only on the inputs?